### DISCUSSION TITLE
What does this code mean? I need an explanation

### DISCUSSION THREAD

**Discussion by srdjanmutlak** [cypress-io/cypress#23174] [Questions and Help]
![explainPLS1](https://user-images.githubusercontent.com/84793610/183243104-147cca1a-4543-4887-adda-d9d0d051d032.jpg)

Can someone please explain how this code works? 

I understand that it's a page object section code. I don't understand why would someone create this kind of code? What would happened if "false" was "true"? Would Cypress then go automatically into the part after ":" ?

Please explain if you saw this before somewhere or if you understand the syntax just by looking at it. 

Here's where the full code is from:
https://github.com/srdjanmutlak/Vivify-QA-task/blob/main/Cypress%20testing/cypress/page_objects/createGalleryPage.js

**Comment by ChriddyN** [comment]
Hi,
the question mark is a shortcut for "if" - for example, for the code about entering the description, it would read like:

```
if (description) {
   this.getInputField('description').clear().type(description)
} else {
   this.getInputField('description').clear()
}
```
